iceoryx2-bb/log/src/lib.rs

```diff
@@ -159,9 +161,14 @@ static DEFAULT_LOGGER: logger::tracing::Logger = logger::tracing::Logger::new();
 static DEFAULT_LOGGER: logger::log::Logger = logger::log::Logger::new();
 
 #[cfg(not(any(feature = "logger_log", feature = "logger_tracing")))]
-static DEFAULT_LOGGER: logger::console::Logger = logger::console::Logger::new();
+pub static DEFAULT_LOGGER: Lazy<logger::console::Logger> =
+    Lazy::new(logger::console::Logger::new);
 
 const DEFAULT_LOG_LEVEL: u8 = LogLevel::Info as u8;
+pub static ENV_LOG_LEVEL: Lazy<LogLevel> = Lazy::new(|| {
+    let log_level_str = env::var("IOX2_LOG_LEVEL").unwrap_or_else(|_| "Info".to_string());
+    LogLevel::from_str(&log_level_str) // Default to Info
+});
```

Comment:
1. Can you explain more about string allocation? how do you track/bench them?
2. about the scenario you mentioned, is not it what should be in the first place? since we might run different instance of application from or within the same shell/env? so each application will utilize its logger?

Reply:
1. There would be [heaptrack](https://apps.kde.org/de/heaptrack/) as a nice gui to track all heap allocation. But in this case I just know that `String` will allocate on the heap from its documentation.
2. Each application will utilize its own logger instance but if the reading of the env variable is delayed, application A might read the value which was set after it was started and meant to be the value for application B.

Reply:
@zmostafa can you please revert the lazy initialization for the logger

Reply:
@elBoberido I reverted back the DEFAULT_LOGGER lazy initialization

Reply:
Can you do the same for `ENV_LOG_LEVEL`